app/rust/src/parser/spending_condition.rs

```diff
@@ -389,7 +396,14 @@ impl<'a> TransactionSpendingCondition<'a> {
             // append the signatures count at the end
             if let Some(count) = self.required_signatures() {
                 buf[20..STANDARD_MULTISIG_AUTH_LEN].copy_from_slice(&count.to_be_bytes());
-                return Ok(STANDARD_MULTISIG_AUTH_LEN);
+                if let Some(b) = buf.get_mut(20..STANDARD_MULTISIG_AUTH_LEN) {
```

Comment:
I think here you forgot to remove the line above... anyways, same as with the others, prefer `.apdu_unwrap` instead of if let so a branch can be removed entirely (well unless you aren't sure it's gonna fit! but you checked that already)... and then copy_from_slice + length check

Also consider using [fill](https://doc.rust-lang.org/std/primitive.slice.html#method.fill) for the fills above (so it becomes roughly `buf[..xyz].fill(0)`, which is much nicer).

Reply:
:+1:

Reply:
seems that fill increases by a few bytes, the size of the app each change.

Reply:
it is now 128 bytes bigger after using fill, run multiple times with/without. same result
although it should be depreciable compared to panic as abort.